packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -1008,7 +1008,10 @@ export class LocalDetachedFluidDataStoreContext
         registry: IProvideFluidDataStoreFactory,
         dataStoreChannel: IFluidDataStoreChannel) {
         assert(this.detachedRuntimeCreation, 0x154 /* "runtime creation is already attached" */);
+        this.detachedRuntimeCreation = false;
```

Comment:
This looks fine, but I'd add try/catch and rejection of this.channelDeferred if something goes wrong.
It was already an issue with bindRuntime() potentially failing, but the scope got even bigger.
If we fail and do not reject this.channelDeferred, then I think some processes may deadlock.

Is there a way to get rid of Deferred and move to regular promises?
In other places I was able to do so and code became more readable (though it needs to be done very carefully)

Via something like

> this.channelPromise = Promise.resolve().then(() => { ... some activity that we want to track });

I.e. ensuring that we create promise synchronously before single line of that activity runs.

For synchronous flow, it would require something like

> this.channelPromise = new Promise((resolve) => {
 ... a bunch of action;
  resolve();
});

i.e. creation of the promise after the fact but capturing success / failure.

Reply:
I get the gist of what you're saying but I think I'll create a task in ADO to track it. Anyone more familiar with this part of the framework can take it on, or I might get to it if I find some time in the coming sprints.